brian2/codegen/runtime/weave_rt/templates/spatialstateupdate.cpp

```diff
@@ -5,22 +5,28 @@
 {# USES_VARIABLES { Cm, dt, v, N,
                   ab_star0, ab_star1, ab_star2, b_plus,
                   ab_plus0, ab_plus1, ab_plus2, b_minus,
-                  ab_minus0, ab_minus1, ab_minus2, v_star, u_plus, u_minus} #}
+                  ab_minus0, ab_minus1, ab_minus2, v_star, u_plus, u_minus,
+                  _P, _B, _morph_i, _morph_parent_i, _starts, _ends,
+                  _invr0, _invrn} #}
 
 {% macro main() %}
     {{ common.insert_group_preamble() }}
 
 	double *_gtot_all=(double *)malloc(N*sizeof(double));
-	double *c=(double *)malloc(N*sizeof(double));
+	double *c=(double *)calloc(N, sizeof(double));
 	double ai,bi,_m;

```

Comment:
Hmm, actually since this `N` is a constant, we could simply declare arrays instead of pointers and wouldn't have to care about the memory management. Maybe that's the simplest solution.